File: CryptoBridge-node/src/presentation/routes/MetricsRouter.js
const express = require('express');
const Joi = require('joi');

const router = express.Router();

// Validation schema
const limitSchema = Joi.object({
    limit: Joi.number().integer().min(1).max(50).default(10)
});

class MetricsRouter {
    constructor(metricsService) {
        this.metricsService = metricsService;
        this.setupRoutes();
    }

    setupRoutes() {
        // GET /metrics/exchanges/top - Top exchanges por número de requisições
        router.get('/exchanges/top', async (req, res) => {
            try {
                const { error, value } = limitSchema.validate({ limit: req.query.limit });
                if (error) {
                    return res.status(422).json({
                        status: 422,
                        msg: 'Parâmetro limit inválido.',
                        details: error.details
                    });
                }

                const result = await this.metricsService.getTopExchanges(value.limit);
                res.json(result);
            } catch (error) {
                console.error('Erro detalhado ao buscar top exchanges:', error);
                res.status(500).json({
                    status: 500,
                    msg: 'Erro interno do servidor.'
                });
            }
        });

        // GET /metrics/tickers/top - Top tickers por número de requisições
        router.get('/tickers/top', async (req, res) => {
            try {
                const { error, value } = limitSchema.validate({ limit: req.query.limit });
                if (error) {
                    return res.status(422).json({
                        status: 422,
                        msg: 'Parâmetro limit inválido.',
                        details: error.details
                    });
                }

                const result = await this.metricsService.getTopTickers(value.limit);
                res.json(result);
            } catch (error) {
                console.error('Erro detalhado ao buscar top tickers:', error);
                res.status(500).json({
                    status: 500,
                    msg: 'Erro interno do servidor.'
                });
            }
        });

        // GET /metrics/performance - Métricas de performance da API
        router.get('/performance', async (req, res) => {
            try {
                const result = await this.metricsService.getPerformanceMetrics();
                res.json(result);
            } catch (error) {
                console.error('Erro detalhado ao buscar métricas de performance:', error);
                res.status(500).json({
                    status: 500,
                    msg: 'Erro interno do servidor.'
                });
            }
        });

        // GET /metrics/summary - Resumo geral das métricas
        router.get('/summary', async (req, res) => {
            try {
                const result = await this.metricsService.getSummary();
                res.json(result);
            } catch (error) {
                console.error('Erro detalhado ao buscar resumo de métricas:', error);
                res.status(500).json({
                    status: 500,
                    msg: 'Erro interno do servidor.'
                });
            }
        });
    }

    getRouter() {
        return router;
    }
}

module.exports = MetricsRouter;

File: CryptoBridge-python/src/presentation/api/routers/metrics_router.py
from fastapi import APIRouter, Depends, Query
from ...schemas.metrics_schemas import (
    TopExchangesResponse, TopTickersResponse,
    PerformanceResponse, SummaryResponse
)
from ..dependencies import get_metrics_service
from ....application.services.metrics_service import MetricsService

router = APIRouter()


@router.get("/exchanges/top", response_model=TopExchangesResponse)
async def get_top_exchanges(
    limit: int = Query(10, ge=1, le=50, description="Número de exchanges no ranking"),
    metrics_service: MetricsService = Depends(get_metrics_service)
):
    try:
        result = await metrics_service.get_top_exchanges(limit)
        return result
    except Exception as e:
        print(f"Erro detalhado ao buscar top exchanges: {str(e)}")
        return {
            "status": 500,
            "msg": "Erro interno do servidor."
        }


@router.get("/tickers/top", response_model=TopTickersResponse)
async def get_top_tickers(
    limit: int = Query(10, ge=1, le=50, description="Número de tickers no ranking"),
    metrics_service: MetricsService = Depends(get_metrics_service)
):
    try:
        result = await metrics_service.get_top_tickers(limit)
        return result
    except Exception as e:
        print(f"Erro detalhado ao buscar top tickers: {str(e)}")
        return {
            "status": 500,
            "msg": "Erro interno do servidor."
        }


@router.get("/performance", response_model=PerformanceResponse)
async def get_performance_metrics(
    metrics_service: MetricsService = Depends(get_metrics_service)
):
    try:
        result = await metrics_service.get_performance_metrics()
        return result
    except Exception as e:
        print(f"Erro detalhado ao buscar métricas de performance: {str(e)}")
        return {
            "status": 500,
            "msg": "Erro interno do servidor."
        }


@router.get("/summary", response_model=SummaryResponse)
async def get_metrics_summary(
    metrics_service: MetricsService = Depends(get_metrics_service)
):
    try:
        result = await metrics_service.get_summary()
        return result
    except Exception as e:
        print(f"Erro detalhado ao buscar resumo de métricas: {str(e)}")
        return {
            "status": 500,
            "msg": "Erro interno do servidor."
        }